R/zchunk_LA1321.cement.R

```diff
@@ -52,72 +63,369 @@ module_energy_LA1321.cement_DISABLED <- function(command, ...) {
     L132.in_EJ_R_indenergy_F_Yh <- get_data(all_data, "L132.in_EJ_R_indenergy_F_Yh")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # 2. Perform computations
+
+    # Set constants used for this chunk
+    # ---------------------------------
+
+    # Extract carbon coefficient for limestone from assumption file
+    LIMESTONE_CCOEF <- A_PrimaryFuelCCoef$PrimaryFuelCO2Coef[A_PrimaryFuelCCoef$PrimaryFuelCO2Coef.name == "limestone"]
+    # Determine historical years not available in data set (additional years) to be copy values from final available year (final_CO2_year)
+    ADDITIONAL_YEARS <- HISTORICAL_YEARS[!HISTORICAL_YEARS %in% energy.CDIAC_CO2_HISTORICAL_YEARS]
+    FINAL_CO2_YEAR <- energy.CDIAC_CO2_HISTORICAL_YEARS[length(energy.CDIAC_CO2_HISTORICAL_YEARS)]
+    # Sets maximum for IO coefficient calculated below
+    MAX_IOELEC <- 4
+
+    # =======================================================================================
+    # Derivation of cement production and limestone consumption by region and historical year
+    # =======================================================================================
+
+    # Downscale Worrell's process CO2 emissions and cement production in 1994 to country level using CDIAC emissions inventory
+    # ------------------------------------------------------------------------------------------------------------------------
+    cement_regions %>%
+      left_join(L100.CDIAC_CO2_ctry_hist, by = "iso") %>%
+      filter(year == 1994) %>%
+      select(iso, Worrell_region, year, cement) %>%
+      na.omit() %>%
+      rename(process_emissions_ktC = cement) ->
+      L1321.Cement_Worrell_ctry
+
+    # Aggregate downscaled process emissions to regional level
+    L1321.Cement_Worrell_ctry %>%
+      group_by(Worrell_region) %>%
+      summarise(reg_process_emissions = sum(process_emissions_ktC)) %>%
+      ungroup() ->
+      L1321.Cement_Worrell_reg
+
+    # Select country-level emissions to prepare for scaling
+    Worrell_1994_cement %>%
+      select(Country, cement_prod_Mt, process_emissions_MtC) ->
+      Worrell_1994_cement_slim
```

Comment:
Couldn't you just do the select when you are joining it later?

Reply:
makes the left_join a bit messier but yea, went for it.